lib/drivers/oneTouchUltra2.js

```diff
@@ -0,0 +1,386 @@
+    } else {
+      return null;
+    }
+  };
+
+  var otu2ErrorHandler = function(info) {
+    // we don't care too much for timeout errors
+    if(info.error && info.error!=='timeout'){
+      console.log('DEBUG OTU2ERROR:', info);
+    }
+    if (info.connectionId && info.error) {
+      if (info.error == 'timeout') {
+        return -9; // TODO check this
+      }
+    }
+  };
+
+  var byteConcat = function (a, b){
+      var aLength = a.length,
+      result = new Uint8Array(aLength + b.length);
+      result.set(a);
+      result.set(b, aLength);
+      return result;
+  };
+
+  var listenForPacket = function (timeout, callback) {
+    //console.log('DEBUG listenForPacket');
+    var abortTimer = setTimeout(function () {
+      clearInterval(listenTimer);
+      console.log('TIMEOUT');
+      callback('TIMEOUT', null);
+    }, timeout);
+
+    var listenTimer = setInterval(function () {
+      var wholebytes = new Uint8Array([0]), pkt;
+
+      while(serialDevice.hasAvailablePacket()) {
+        pkt = serialDevice.nextPacket();
+        wholebytes = byteConcat(wholebytes, pkt.bytes);
+      }
+      pkt = buildPacket(wholebytes);
+      var tostr = String.fromCharCode.apply(String, pkt.bytes);
+      console.log('DEBUG OTU2 listenForPacket:', pkt, tostr);
+      clearTimeout(abortTimer);
+      clearInterval(listenTimer);
+
+      callback(null, pkt);
+    }, 1000);     // spin on this one not so quickly
+  };
+
+  var probe = function(cb){
+    console.log('attempting probe of oneTouch Ultra2 - don\'t probe anything');
+    //otu2CommandResponse(DM, cb);
+    cb();
+  };
+
+  var otu2CommandResponse = function (command, callback) {
+    var cmd = struct.packString(command);
+    serialDevice.writeSerial(cmd, function () {
+      listenForPacket(18000, function(err, pkt) {
+        if (err === 'TIMEOUT') {
+        } else {
+          callback(err, pkt);
+        }
+      });
+
+    });
+
+  };
+
+  var getSomeInfo = function (obj, cb) {
+      console.log('DEBUG: on getSomeInfo');
+      cb();
+  };
+
+  var prepBGData = function (progress, data) {
+    cfg.builder.setDefaults({ deviceId: data.id });
+    var dataToPost = [];
+
+    //This is converting to a string the data bytes array
+    var lines = String.fromCharCode.apply(null, data.bytes);
+
+    //Separe each data row by \n
+    var splites = lines.split('\n');
+
+    // extract header
+    var hpattern = /P (\d{3}),"(\w{9})","(\w+\/\w+).*" (\w+)/;
+
+    var toMatch = hpattern.exec(splites[0]);
+    var nrecords = toMatch[1],
+        serialno = toMatch[2],
+        oum = toMatch[3],
+        checksum = toMatch[4],
+        calcchks = 0;
+
+    //set missing serial number and id
+    data.serialNumber = serialno;
+    data.id = data.id + serialno;
+
+    // var bytes = data.bytes.subarray(0, 27, 0);
+    var stringWithoutChk = 'P ' + nrecords + ',"' + serialno + '","' + oum + ' "';
+    var bytes = [0];
+
+    for (var i = 0; i < stringWithoutChk.length; ++i)
```

Comment:
I'm sorry to insist, but this is really the wrong way to do it. The whole point of a checksum is to validate the actual bytes received to make sure that what you receive is what was sent. Doing this by parsing and then reassembling the string goes against the entire idea. Please write a loop over the original data to calculate the checksum.